lib/membrane_rtc_engine/endpoints/hls_endpoint.ex

```diff
@@ -294,26 +297,50 @@ if Enum.all?(
         manifest_module: Membrane.HTTPAdaptiveStream.HLS,
         target_window_duration: state.target_window_duration,
         target_segment_duration: state.target_segment_duration,
+        muxer_segment_duration: state.target_segment_duration - Membrane.Time.seconds(1),
         persist?: false,
         storage: %Membrane.HTTPAdaptiveStream.Storages.FileStorage{
           directory: directory
         },
         hls_mode: state.hls_mode
       }
 
+      audio_mixer = %Membrane.AudioMixer{
+        caps: %Membrane.RawAudio{
+          channels: 1,
+          sample_rate: 48_000,
+          sample_format: :s16le
+        }
+      }
+
+      {frames_per_second, 1} = state.compositor_config.output_framerate
+      seconds_number = div(state.target_segment_duration, Membrane.Time.seconds(1))
+
       %ParentSpec{
         children: %{
-          :compositor => compositor,
-          :video_parser_out => video_parser_out,
-          :hls_sink_bin => hls_sink_bin
+          compositor: compositor,
+          video_parser_out: video_parser_out,
+          hls_sink_bin: hls_sink_bin,
+          audio_mixer: audio_mixer,
+          aac_encoder: Membrane.AAC.FDK.Encoder,
+          aac_parser: %Membrane.AAC.Parser{out_encapsulation: :none}
         },
         links: [
           link(:compositor)
+          # |> to(:framerate_converter, %Membrane.FramerateConverter{
+          #   framerate: state.compositor_config.output_framerate
+          # })
           |> to(:encoder, %Membrane.H264.FFmpeg.Encoder{
-            profile: :baseline
+            profile: :baseline,
+            gop_size: frames_per_second * seconds_number
           })
           |> to(:video_parser_out)
           |> via_in(Pad.ref(:input, :video), options: [encoding: :H264])
+          |> to(:hls_sink_bin),
+          link(:audio_mixer)
+          |> to(:aac_encoder)
+          |> to(:aac_parser)
+          |> via_in(Pad.ref(:input, :audio), options: [encoding: :AAC])
```

Comment:
Can the audio mixer part be separated from a video part of this function? I feel the function `compose_hls_sink_bin` could be written on a higher abstraction level.